driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -86,38 +94,55 @@ public void run() {
             }
         };
 
+        this.connections = new CopyOnWriteArrayList<PooledConnection>();
+        this.open = new AtomicInteger();
+        this.initFuture = SettableFuture.create();
+    }
+
+    public void initAsync() {
```

Comment:
A minor enhancement would be to make this method return initFuture (cf. similar methods such as closeAsync) and make initFuture field private.